src/admin/components/elements/DuplicateDocument/index.tsx

```diff
@@ -1,39 +1,143 @@
-  const { getData } = useForm();
+  const modified = useFormModified();
+  const { toggle } = useModal();
+  const { setModified } = useForm();
+  const { serverURL, routes: { api }, localization } = useConfig();
   const { routes: { admin } } = useConfig();
+  const [hasClicked, setHasClicked] = useState<boolean>(false);
 
-  const handleClick = useCallback(() => {
-    const data = getData();
+  const modalSlug = `duplicate-${id}`;
 
-    push({
-      pathname: `${admin}/collections/${slug}/create`,
-      state: {
-        data,
-      },
-    });
-  }, [push, getData, slug, admin]);
+  const handleClick = useCallback(async (override = false) => {
+    setHasClicked(true);
+
+    if (modified && !override) {
+      toggle(modalSlug);
+      return;
+    }
+
+    const create = async (locale?: string): Promise<string | null> => {
+      const localeParam = locale ? `locale=${locale}` : '';
+      const response = await requests.get(`${serverURL}${api}/${slug}/${id}?${localeParam}`);
+      const data = await response.json();
+      const result = await requests.post(`${serverURL}${api}/${slug}`, {
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(data),
+      });
+      const json = await result.json();
+
+      if (result.status === 201) {
+        return json.doc.id;
+      }
+      json.errors.forEach((error) => toast.error(error.message));
+      return null;
+    };
+
+    let duplicateID;
+    if (localization) {
+      duplicateID = await create(localization.defaultLocale);
+      let abort = false;
+      localization.locales
+        .filter((locale) => locale !== localization.defaultLocale)
+        .forEach(async (locale) => {
+          if (!abort) {
+            const res = await requests.get(`${serverURL}${api}/${slug}/${id}?locale=${locale}`);
+            const localizedDoc = await res.json();
+            const patchResult = await requests.patch(`${serverURL}${api}/${slug}/${duplicateID}?locale=${locale}`, {
+              headers: {
+                'Content-Type': 'application/json',
+              },
+              body: JSON.stringify(localizedDoc),
+            });
+            if (patchResult.status > 400) {
+              abort = true;
+              const json = await patchResult.json();
+              json.errors.forEach((error) => toast.error(error.message));
+            }
+          }
+        });
+      if (abort) {
+        // delete the duplicate doc to prevent incomplete
+        await requests.delete(`${serverURL}${api}/${slug}/${id}`);
+      }
+    } else {
+      duplicateID = await create();
+    }
+
+    toast.success(`${collection.labels.singular} successfully duplicated.`,
+      { autoClose: 3000 });
+
+    const previousModifiedState = modified;
+    setModified(false);
+    setTimeout(() => {
+      push({
+        pathname: `${admin}/collections/${slug}/${duplicateID}`,
+      });
+      setModified(previousModifiedState);
```

Comment:
I don't think we need this, right? because the `push` will have taken us to a new `edit` view entirely, and the form should NOT be modified in that state.

Reply:
I did this because the interaction felt a little buggy. You could duplicate on a modified document, continue through the modal, then the form would show the changes as though they were saved and the UI would let you leave even though the last thing you saw was unsaved.

I opted to preserve the modified state so that it would be super clear to the editor.